conformance/lang/expressions/additive-expr/string_addition.balt

```diff
@@ -0,0 +1,609 @@
+
+function init() {
+    Bar b = {name: "test"};
+    io:println(b.name + (new Foo()).id); // @output testABC
+}
+
+class Foo {
+    string id = "ABC";
+}
+
+type Bar record {|
+    string name;
+|};
+
+Test-Case: output
+Description: Test string concatenation with function call and method call expressions as operands.
+Labels: additive-expr, string, function-call-expr, method-call-expr, implicit-new-expr, explicit-new-expr, module-class-defn
+
+function init() {
+    Class cl = new;
+    io:println(getString() + cl.getString()); // @output foobar
+    io:println(cl.getString() + getString()); // @output barfoo
+    io:println(getString() + getString()); // @output foofoo
+    io:println(cl.getString() + (new Class()).getString()); // @output barbar
+}
+
+function getString() returns string {
+    return "foo";
+}
+
+class Class {
+    function getString() returns string {
+        return "bar";
+    }
+}
+
+Test-Case: output
+Description: Test string concatenation with let expressions as operands.
+Labels: additive-expr, string, let-expr
+
+function init() {
+    string s = (let string a = "hello", string b = "world" in a + " " + b ) + let string a = "!" in a;
+    io:println(s); // @output hello world!
+}
+
+Test-Case: output
+Description: Test string concatenation with type cast expressions as operands.
+Labels: additive-expr, string, type-cast-expr, optional-type, string:Char
+
+function init() {
+    string s = "!";
+    string? t = "hi";
+    io:println(<string> t + <string:Char> s); // @output hi!
+}
+
+Test-Case: output
+Description: Test string concatenation with additive expressions as operands.
+Labels: additive-expr, string, module-type-defn, union-type, singleton-type, string:Char
+
+type Chars "H"|"I";
+
+function init() {
+    Chars s = "H";
+    string:Char t = "e";
+    string u = "llo";
+    string v = " from ballerina";
+    string w = "!";
+
+    string x = (s + t + u) + v + w;
+    io:println(x); // @output Hello from ballerina!
+
+    string y = s + t + u + v + w;
+    io:println(y); // @output Hello from ballerina!
+
+    string z = (s + t) + u + v + w;
+    io:println(z); // @output Hello from ballerina!
+}
+
+Test-Case: output
+Description: Test string concatenation with conditional expressions as operands.
+Labels: additive-expr, string, union-expr, optional-type, error, conditional-expr
+
+function init() {
+    string? s = "hello";
+    string|error t = " world";
+    string u = (s ?: "hi") + (t is error ? "!" : t);
+    io:println(u); // @output hello world
+}
+
+Test-Case: output
+Description: Test string concatenation with checking expressions as operands.
+Labels: additive-expr, string, union-expr, optional-type, error, check, checkpanic
+
+function init() returns error? {
+    string|error s = "hello";
+    string|error t = " world";
+    string u = check s + checkpanic t;
+    io:println(u); // @output hello world
+    io:println(check s + checkpanic t); // @output hello world
+}
```

Comment:
Shall we also test string/xml concatenation at module level? And valid assignments with `var`?